src/Features/VisualBasic/Portable/Completion/CompletionProviders/ObjectCreationCompletionProvider.vb

```diff
@@ -51,5 +51,20 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.Providers
         Protected Overrides Function GetCompletionItemRules(symbols As IReadOnlyList(Of ISymbol), preselect As Boolean) As CompletionItemRules
             Return s_rules
         End Function
+
+        Protected Overrides Function GetDisplayAndInsertionText(symbol As ISymbol, context As SyntaxContext) As (displayText As String, insertionText As String)
+            ' If not all  of the type arguments are concrete types, eg
+            ' we've inferred something like `Task(Of T)`, just suggest the
+            ' generic `Task(Of ...)`
+            Dim enclosingSymbol = context.SemanticModel.GetEnclosingSymbol(context.Position)
+            Dim nt = TryCast(symbol, INamedTypeSymbol)
+            If nt.TypeArguments.Any(Function(t) t.TypeKind = TypeKind.TypeParameter AndAlso
+                                            Not DirectCast(t, ITypeParameterSymbol).IsAccessibleFromSymbolOrEnclosingType(enclosingSymbol)) Then
```

Comment:
1. all my comments about c# apply here.
2. if your logic is identical, we should find a way to share this code between C# and VB in some helper method.